EmitC ttnn.mish error

### Describe the bug

There is an emitc compile error for ttnn.mish. It wasn't discovered previously because builder emitc tests were skipped due to a pytest hang.

[CI failure](https://github.com/tenstorrent/tt-mlir/actions/runs/23465343932/job/68277709299#step:16:6693)

```
        except subprocess.CalledProcessError as e:
>           raise EmitCCompileError(
                f"Build failed for {cpp_base_name}:\nstdout: {e.stdout}\nstderr: {e.stderr}"
            )
E           emitc_compiler.EmitCCompileError: Build failed for emitc_compiled:
E           stdout: [1/2] Building CXX object CMakeFiles/ttnn-dylib.dir/ttnn-dylib.cpp.o
E           FAILED: [code=1] CMakeFiles/ttnn-dylib.dir/ttnn-dylib.cpp.o 
E           /usr/bin/clang++ -Dttnn_dylib_EXPORTS -I/__w/tt-mlir/tt-mlir/install/tt-metal/.cpmcache/reflect/f93e77475670eaeacf332927dfe8b50e3f3812e0 -I/__w/tt-mlir/tt-mlir/install/tt-metal/.cpmcache/fmt/69912fb6b71fcb1f7e5deca191a2bb4748c4e7b6/include -I/__w/tt-mlir/tt-mlir/install/tt-metal/.cpmcache/enchantum/2fb7ab238e36c101b9848892ddb6382276b65837/enchantum/include -I/__w/tt-mlir/tt-mlir/install/tt-metal/.cpmcache/boost/1359e136761ab2d10afa1c4e21086c8d824735cd/libs/core/include -I/__w/tt-mlir/tt-mlir/install/tt-metal/.cpmcache/nlohmann_json/798e0374658476027d9723eeb67a262d0f3c8308/include -I/__w/tt-mlir/tt-mlir/install/tt-metal/.cpmcache/spdlog/b1c2586bb5c35a7929362e87f62433eb68206873/include -I/__w/tt-mlir/tt-mlir/install/tt-metal/.cpmcache/tt-logger/d2339ce68562cae34cd95f3fece7fd94eb0529b7/include -I/__w/tt-mlir/tt-mlir/install/tt-metal -I/__w/tt-mlir/tt-mlir/install/tt-metal/tt_metal -I/__w/tt-mlir/tt-mlir/install/tt-metal/tt_metal/api -I/__w/tt-mlir/tt-mlir/install/tt-metal/tt_metal/hostdevcommon/api -I/__w/tt-mlir/tt-mlir/install/tt-metal/tt_metal/hw/inc -I/__w/tt-mlir/tt-mlir/install/tt-metal/tt_metal/hw/inc/wormhole/wormhole_b0_defines -I/__w/tt-mlir/tt-mlir/install/tt-metal/tt_metal/hw/inc/wormhole -I/__w/tt-mlir/tt-mlir/install/tt-metal/tt_metal/hw/inc/blackhole -I/__w/tt-mlir/tt-mlir/install/tt-metal/tt_metal/include -I/__w/tt-mlir/tt-mlir/install/tt-metal/tt_stl -I/__w/tt-mlir/tt-mlir/install/tt-metal/tt_stl/tt_stl -I/__w/tt-mlir/tt-mlir/install/tt-metal/tt_metal/third_party/fmt -I/__w/tt-mlir/tt-mlir/install/tt-metal/tt_metal/third_party/enchantum -I/__w/tt-mlir/tt-mlir/install/tt-metal/tt_metal/third_party/taskflow -I/__w/tt-mlir/tt-mlir/install/tt-metal/tt_metal/third_party/tracy/public -I/__w/tt-mlir/tt-mlir/install/tt-metal/tt_metal/third_party/umd -I/__w/tt-mlir/tt-mlir/install/tt-metal/tt_metal/third_party/umd/device/api -I/__w/tt-mlir/tt-mlir/install/tt-metal/tt_metal/third_party/umd/src/firmware/riscv/wormhole -I/__w/tt-mlir/tt-mlir/install/tt-metal/tt_metal/third_party/umd/src/firmware/riscv/blackhole -I/__w/tt-mlir/tt-mlir/install/tt-metal/ttnn -I/__w/tt-mlir/tt-mlir/install/tt-metal/ttnn/api -I/__w/tt-mlir/tt-mlir/install/tt-metal/ttnn/core -I/__w/tt-mlir/tt-mlir/install/tt-metal/ttnn/cpp -I/__w/tt-mlir/tt-mlir/install/tt-metal/ttnn/cpp/ttnn -O3 -DNDEBUG -std=gnu++20 -fPIC -march=x86-64-v3 -Winvalid-pch -Xclang -include-pch -Xclang /__w/tt-mlir/tt-mlir/tools/ttnn-standalone/build/CMakeFiles/ttnn-dylib.dir/cmake_pch.hxx.pch -Xclang -include -Xclang /__w/tt-mlir/tt-mlir/tools/ttnn-standalone/build/CMakeFiles/ttnn-dylib.dir/cmake_pch.hxx -MD -MT CMakeFiles/ttnn-dylib.dir/ttnn-dylib.cpp.o -MF CMakeFiles/ttnn-dylib.dir/ttnn-dylib.cpp.o.d -o CMakeFiles/ttnn-dylib.dir/ttnn-dylib.cpp.o -c /__w/tt-mlir/tt-mlir/tools/ttnn-standalone/ttnn-dylib.cpp
E           /__w/tt-mlir/tt-mlir/tools/ttnn-standalone/ttnn-dylib.cpp:4:23: error: no matching function for call to object of type 'const registered_operation_t<reflect::fixed_string<char, 10UL>{"ttnn::mish"}, ExecuteUnaryWithFastAndApproximateMode<ttnn::operations::unary::UnaryOpType::MISH>>'
E               4 |   ::ttnn::Tensor v3 = ttnn::mish(v2, ::ttnn::MemoryConfig{::ttnn::TensorMemoryLayout::INTERLEAVED, ::ttnn::BufferType::DRAM, ::std::nullopt});
E                 |                       ^~~~~~~~~~
E           /__w/tt-mlir/tt-mlir/install/tt-metal/ttnn/api/ttnn/decorators.hpp:79:10: note: candidate template ignored: constraints not satisfied [with Args = <::ttnn::Tensor &, ::ttnn::MemoryConfig>]
E              79 |     auto operator()(Args&&... args) const {
E                 |          ^
E           /__w/tt-mlir/tt-mlir/install/tt-metal/ttnn/api/ttnn/decorators.hpp:78:18: note: because 'HasInvoke<ttnn::operations::unary::ExecuteUnaryWithFastAndApproximateMode<ttnn::operations::unary::UnaryOpType::MISH>, tt::tt_metal::Tensor &, tt::tt_metal::MemoryConfig &&>' evaluated to false
E              78 |         requires(HasInvoke<operation_t, Args && ...>)
E                 |                  ^
E           /__w/tt-mlir/tt-mlir/install/tt-metal/ttnn/api/ttnn/decorators.hpp:61:18: note: because 'Op::invoke(std::declval<Args>()...)' would be invalid: no viable conversion from 'tt::tt_metal::MemoryConfig' to 'bool'
E              61 |     { Op::invoke(std::declval<Args>()...) };
E                 |                  ^
E           1 error generated.
E           ninja: build stopped: subcommand failed.
```

### Steps to reproduce the issue

To repro: 

remove skip from test in `test_ttir_unary.py` and run
```
pytest test/python/golden/ttir_ops/eltwise/test_ttir_unary.py::test_unary_ops[mish-emitc-f32-128x128]
```

### Expected behavior

_No response_

### Please complete the following environment information

_No response_

--- a/lib/Conversion/TTNNToEmitC/TTNNToEmitC.cpp
+++ b/lib/Conversion/TTNNToEmitC/TTNNToEmitC.cpp
@@ -4818,7 +4818,8 @@ void populateTTNNToEmitCPatterns(mlir::MLIRContext *ctx,
            EltwiseUnaryOpConversionPattern<mlir::tt::ttnn::Relu6Op>,
            EltwiseUnaryOpConversionPattern<mlir::tt::ttnn::HardsigmoidOp>,
            EltwiseUnaryOpConversionPattern<mlir::tt::ttnn::SiluOp>,
-           EltwiseUnaryOpConversionPattern<mlir::tt::ttnn::MishOp>,
+           EltwiseUnaryWithFastAndApproximateModeOpConversionPattern<
+               mlir::tt::ttnn::MishOp>,
            ElementwiseUnaryWithFloatParameterOpConversionPattern<
                mlir::tt::ttnn::LeakyReluOp>,
            EltwiseUnaryWithFastAndApproximateModeOpConversionPattern<

--- a/test/python/golden/ttir_ops/eltwise/test_ttir_unary.py
+++ b/test/python/golden/ttir_ops/eltwise/test_ttir_unary.py
@@ -298,9 +298,6 @@ def test_unary_ops(
     ]:
         pytest.skip("int32 unary op is not supported yet for this operation")
 
-    if target == "emitc" and test_fn in [mish]:
-        pytest.skip("https://github.com/tenstorrent/tt-mlir/issues/7654")
-
     def module(builder: TTIRBuilder):
         @builder.func([shape], [dtype])
         def unary_op(

